File: java/lesson 1/Shuffle.java
// https://leetcode.com/problems/shuffle-an-array/

class Solution {
    public void shuffle(int[] values) {
        Random random = new Random();
        for (int i = 0; i < values.length; i++) {
            int r = random.nextInt(i + 1);
            int temp = values[i];
            values[i] = values[r];
            values[r] = temp;
        }
    }
}

File: javascript/lesson 1/shuffle.js
// https://leetcode.com/problems/shuffle-an-array/

function shuffle(values) {
    for (let i = 0; i < values.length; i++) {
        const randIndex = Math.floor(Math.random() * (i + 1));
        [values[randIndex], values[i]] = [values[i], values[randIndex]];
    }
}
